spec/host-interpolate-by-header/run_spec.lua

```diff
@@ -277,13 +281,164 @@ for _, strategy in helpers.each_strategy() do
+      )
+
+      describe(
+        "\n ** Should fail with status 502 as the port provided in config is an invalid port",
+        function()
+          -- Update plugin config via admin_client
+          admin_client = helpers.admin_client(60000)
+          local url = "/plugins/" .. host_interpolate_by_header["id"]
+          assert(admin_client:patch(
+            url,
+            {
+              headers = {
+                ["Content-Type"] = "application/json"
+              },
+              body = {
+                name = "host-interpolate-by-header",
+                config = {
+                  host = "<"..  host_placeholder ..">",
+                  fallback_host = "",
+                  headers = {host_placeholder},
+                  port = 10000,   --invalid port
+                  operation = "none",
+                  modulo_by = 1,
+                },
+                enabled = true,
+                consumer = nil
+              }
+            }
+          ))
+
+          proxy_client = helpers.proxy_client()
+          local res = assert(proxy_client:send(
+            {
+              method = "GET",
+              path = "/gethost",
+              headers = {
+                ["Content-type"] = "application/json",
+                [host_placeholder] = SERVER_IP,
+              },
+              data = {}
+            }
+          ))
+
+          it(
+            "\nStatus code should be 502",
+            function()
+              assert(res.status == 502)
+            end
+          )
+        end
+      )
+
+      describe(
+        "\n ** Should replace place_holder with a hyphen in the host",
+        function()
+          -- Update plugin config via admin_client
+          admin_client = helpers.admin_client(60000)
+          local url = "/plugins/" .. host_interpolate_by_header["id"]
+          assert(admin_client:patch(
+            url,
+            {
+              headers = {
+                ["Content-Type"] = "application/json"
+              },
+              body = {
+                name = "host-interpolate-by-header",
+                config = {
+                  host = "service_<".. host_placeholder ..">.com",
+                  fallback_host = "",
+                  headers = {host_placeholder},
+                  operation = "none",
+                  modulo_by = 1,
+                },
+                enabled = true,
+                consumer = nil
+              }
+            }
+          ))
+
+          proxy_client = helpers.proxy_client()
+          local res = assert(proxy_client:send(
+            {
+              method = "GET",
+              path = "/gethost",
+              headers = {
+                ["Content-type"] = "application/json",
+                [host_placeholder] = "abc_xyz",
+              },
+              data = {}
+            }
+          ))
+
+          it(
+            "\nStatus code should be 200",
+            function()
+              assert(res.status == 200)
```

Comment:
assert for host also to check whether request is proxied to correct URL or not

Reply:
Done